### ISSUE TITLE
Code block deletes whitespace at start of first line

### ISSUE THREAD

**Issue by johnny-smitherson** [hyperdiv/hyperdiv#17]
```
import hyperdiv as hd
import pandas as pd

df = pd.DataFrame( {'K': [1, 2, 3], 'V': [4, 5, 6], 'X': ['a', 'b', 'c']})
hd.code(str(df))
hd.markdown('```\n' + str(df) + '\n```')
```

<img width="207" alt="Cap123123ture" src="https://github.com/hyperdiv/hyperdiv/assets/127537716/e3f229e1-77c8-4b5c-9449-4a500592cf64">

Second one (markdown ``` block) is the correct whitespace

At first look, the `code()` function should output in my mind a `<code>` block - but inspecting the implementation I see  by default `language=python`. That explains the funny colors.

Providing `hd.code(str(df), language='')` works as expected - I guess this would be a question of defaults. In my opinion, people will be using code blocks for walls of numbers/jsons more often than actual python code - but if you write a documentation site, it makes sense to say all code blocks are python - why not have a special `hd.python()` for that?

You could also have some trivial auto-detection for example by trying a bunch of things like `json.loads(x)` and `ast.parse(x)` and `configparser.ConfigParser().read_file(io.StringIO(x))` to see if any of these don't throw exceptions - or use `libmagic` or use some external package

**Comment by mariusnita** [comment]
Thanks, I'll look into this. Currently Markdown (hd.code() calls hd.markdown() internally) blocks are de-dented and stripped: https://github.com/hyperdiv/hyperdiv/blob/main/hyperdiv/components/markdown.py#L39

Probably shouldn't strip() there.

**Comment by johnny-smitherson** [comment]
I don't think the strip affects it, because the `code()` function surrounds it with ``` anyway. Since unmodified `hd.code(str(df), language='')` shows correct whitespace, i think this is only connected to the language highlighter which is turned on for python by default.

**Comment by mariusnita** [comment]
I fixed this, will be in the next release probably tomorrow. The core issue was using `stripall=True` here: https://github.com/hyperdiv/hyperdiv/pull/23/files#diff-1aaa08262c320c7aec58bc720cb5bccd3d32cb9260451569b05f2bcf9da10b37L20

I'm leaving python as the default language in code() for now. You can pass language="text" for text blocks. language="" will default to "text" as well.

**Comment by mariusnita** [comment]
Btw fun fact I just learned: You can render a DataFrame in a table: `hd.data_table(df)`. DataFrame supports a dictionary interface and seems to work out of the box.

**Comment by johnny-smitherson** [comment]
ah, i thought that's why you made the argument be dict of columns - for pandas compatibility

**Comment by mariusnita** [comment]
Fixed in v0.1.2
